packages/esp-js/src/system/health/aggregateHealthIndicator.ts

```diff
@@ -0,0 +1,155 @@
+import {HealthIndicator} from './healthIndicator';
+import {Health, HealthStatus} from './health';
+import {DefaultHealthIndicatorTrigger, HealthIndicatorTrigger} from './healthIndicatorTrigger';
+import {Logger} from '../logging/logger';
+import {DisposableBase} from '../disposables/disposableBase';
+import {Guard} from '../guard';
+
+const _defaultLog = Logger.create('AggregateHealthIndicator');
+
+export class AggregateHealthIndicator extends DisposableBase implements HealthIndicator {
+    private _isStarted = false;
+    private _health = Health.builder(AggregateHealthIndicator.constructor.name).isUnknown().build();
+    private _healthIndicators: WeakRef<HealthIndicator>[] = [];
+
+    constructor(private _healthIndicatorTrigger: HealthIndicatorTrigger = DefaultHealthIndicatorTrigger, private _log: Logger = _defaultLog) {
+        super();
+    }
+
+    public get healthIndicatorName() {
+        return AggregateHealthIndicator.constructor.name;
+    }
+
+    public health(): Health {
+        return this._health;
+    }
+
+    public get activateIndicatorCount() {
+        return this._healthIndicators.length;
+    }
+
+    public addIndicator = (healthIndicator: HealthIndicator) => {
```

Comment:
- [ ] AggregateHealthIndicator -  add an overload that allows the indicator to be hard referenced